main.c

```diff
@@ -0,0 +1,138 @@
+#include <stdio.h>
+#include <string.h>
+
+enum
+{
+    data,
+    next
+};
+
+void StringListInit(char*** list)
+{
+    list[data] = NULL;
+    list[next] = NULL;
+}
+
+void StringListAdd(char*** list, char* str)
+{
+    char** node = (char**)malloc(sizeof(char*) * 2);
+    node[data] = (char*)malloc(sizeof(char) * strlen(str) + next);
+
+    // copy our data to node
+    strcpy((char*)node[data], str);

```

Comment:
The potential null pointer is passed into 'strcpy' function. Inspect the first argument.